### ISSUE TITLE
Seems not to receive SmbAuthException when GUEST login is disabled (SMB1)

### ISSUE THREAD

**Issue by courville** [AgNO3/jcifs-ng#141]
Strangely, with latest git version (and also before), when browsing a share for which explicitly GUEST login is disabled (ASUS router RT-AC68U) and using GUEST login, I still can browse root of SMB server and when accessing subfolder I do not get an SmbAuthException which prevents my app to request for credentials. Below is the log of my app:
```
02-23 19:21:25.599 20012 22285 D JcifsFile2: smb://RT-AC68U/movies/
02-23 19:21:25.602 20012 22285 D JcifsFile2:  SmbException on smb://RT-AC68U/movies/
02-23 19:21:25.602 20012 22285 D JcifsFile2: smb://RT-AC68U/@eaDir/
02-23 19:21:25.606 20012 22285 D JcifsFile2:  SmbException on smb://RT-AC68U/@eaDir/
02-23 19:21:25.606 20012 22285 D JcifsFile2: smb://RT-AC68U/@tmp/
02-23 19:21:25.609 20012 22285 D JcifsFile2:  SmbException on smb://RT-AC68U/@tmp/
02-23 19:21:25.609 20012 22285 D JcifsFile2: smb://RT-AC68U/manga/
02-23 19:21:25.613 20012 22285 D JcifsFile2:  SmbException on smb://RT-AC68U/manga/
02-23 19:21:25.613 20012 22285 D JcifsFile2: smb://RT-AC68U/LOST.DIR/
02-23 19:21:25.616 20012 22285 D JcifsFile2:  SmbException on smb://RT-AC68U/LOST.DIR/
02-23 19:21:25.616 20012 22285 D JcifsFile2: smb://RT-AC68U/Android/
02-23 19:21:25.619 20012 22285 D JcifsFile2:  SmbException on smb://RT-AC68U/Android/
02-23 19:21:28.576 20012 22316 D JcifListingEngine: listFiles for:smb://RT-AC68U/movies/
02-23 19:21:28.596 20012 22316 E JcifListingEngine: JcifListingThread: SmbException
02-23 19:21:28.596 20012 22316 E JcifListingEngine: jcifs.smb.SmbException: Invalid operation for IPC service: Trans2FindFirst2[command=SMB_COM_TRANSACTION2,received=false,errorCode=0,flags=0x0018,flags2=0x0000,signSeq=0,tid=2,pid=55327,uid=0,mid=0,wordCount=0,byteCount=0,totalParameterCount=0,totalDataCount=0,maxParameterCount=10,maxDataCount=65535,maxSetupCount=0,flags=0x00,timeout=0,parameterCount=0,parameterOffset=0,parameterDisplacement=0,dataCount=0,dataOffset=0,dataDisplacement=0,setupCount=1,pad=0,pad1=0,searchAttributes=0x16,searchCount=200,flags=0x00,informationLevel=0x104,searchStorageType=0,filename=\]
02-23 19:21:28.596 20012 22316 E JcifListingEngine:     at jcifs.smb.SmbTreeImpl.checkRequest(SmbTreeImpl.java:519)
02-23 19:21:28.596 20012 22316 E JcifListingEngine:     at jcifs.smb.SmbTreeImpl.send(SmbTreeImpl.java:452)
02-23 19:21:28.596 20012 22316 E JcifListingEngine:     at jcifs.smb.SmbTreeConnection.send0(SmbTreeConnection.java:399)
02-23 19:21:28.596 20012 22316 E JcifListingEngine:     at jcifs.smb.SmbTreeConnection.send(SmbTreeConnection.java:314)
02-23 19:21:28.596 20012 22316 E JcifListingEngine:     at jcifs.smb.SmbTreeConnection.send(SmbTreeConnection.java:294)
02-23 19:21:28.596 20012 22316 E JcifListingEngine:     at jcifs.smb.SmbTreeHandleImpl.send(SmbTreeHandleImpl.java:130)
02-23 19:21:28.596 20012 22316 E JcifListingEngine:     at jcifs.smb.DirFileEntryEnumIterator1.open(DirFileEntryEnumIterator1.java:67)
02-23 19:21:28.596 20012 22316 E JcifListingEngine:     at jcifs.smb.DirFileEntryEnumIteratorBase.<init>(DirFileEntryEnumIteratorBase.java:67)
02-23 19:21:28.596 20012 22316 E JcifListingEngine:     at jcifs.smb.DirFileEntryEnumIterator1.<init>(DirFileEntryEnumIterator1.java:46)
02-23 19:21:28.596 20012 22316 E JcifListingEngine:     at jcifs.smb.SmbEnumerationUtil.doEnum(SmbEnumerationUtil.java:229)
02-23 19:21:28.596 20012 22316 E JcifListingEngine:     at jcifs.smb.SmbEnumerationUtil.listFiles(SmbEnumerationUtil.java:283)
02-23 19:21:28.596 20012 22316 E JcifListingEngine:     at jcifs.smb.SmbFile.listFiles(SmbFile.java:1266)
02-23 19:21:28.596 20012 22316 E JcifListingEngine:     at com.archos.filecorelibrary.jcifs.JcifListingEngine$JcifListingThread.run(JcifListingEngine.java:123)
```
I can do some captures if need be.

**Comment by courville** [comment]
Reverting to the test python script available here https://gist.github.com/courville/a0e6fe1ce2f31c9adc52191216eed3e0 with GUEST login disabled on server, strangely, I can browse the root:
```
java smbcli 2 smb://RT-AC68U/ "" GUEST ""
smb://RT-AC68U/movies/ exists
smb://RT-AC68U/@eaDir/ exists
smb://RT-AC68U/ipc$/ exists but not accessible
smb://RT-AC68U/@tmp/ exists
smb://RT-AC68U/manga/ exists
smb://RT-AC68U/LOST.DIR/ exists
smb://RT-AC68U/Android/ exists
[Thread-1] WARN jcifs.smb.SmbTransportImpl - Disconnecting transport while still in use Transport0[RT-AC68U<20>/192.168.0.27:445,state=5,signingEnforced=false,usage=1]: [SmbSession[credentials=GUEST,targetHost=rt-ac68u,targetDomain=null,uid=100,connectionState=2,usage=18], SmbSession[credentials=,targetHost=rt-ac68u,targetDomain=null,uid=101,connectionState=2,usage=0]]
[Thread-1] WARN jcifs.smb.SmbSessionImpl - Logging off session while still in use SmbSession[credentials=GUEST,targetHost=rt-ac68u,targetDomain=null,uid=100,connectionState=3,usage=18]:[SmbTree[share=IPC$,service=IPC,tid=1,inDfs=false,inDomainDfs=false,connectionState=2,usage=0], SmbTree[share=MOVIES,service=?????,tid=0,inDfs=false,inDomainDfs=false,connectionState=0,usage=0], SmbTree[share=@EADIR,service=?????,tid=0,inDfs=false,inDomainDfs=false,connectionState=0,usage=0], SmbTree[share=@TMP,service=?????,tid=0,inDfs=false,inDomainDfs=false,connectionState=0,usage=0], SmbTree[share=MANGA,service=?????,tid=0,inDfs=false,inDomainDfs=false,connectionState=0,usage=0], SmbTree[share=LOST.DIR,service=?????,tid=0,inDfs=false,inDomainDfs=false,connectionState=0,usage=0], SmbTree[share=ANDROID,service=?????,tid=0,inDfs=false,inDomainDfs=false,connectionState=0,usage=0]]
```
And going into the movies subdir I get a CIFSException:
```
java smbcli 2 smb://RT-AC68U/movies/ "" GUEST ""
Caught a CIFSException on listFiles
jcifs.smb.SmbException: Invalid operation for IPC service: Trans2FindFirst2[command=SMB_COM_TRANSACTION2,received=false,errorCode=0,flags=0x0018,flags2=0x0000,signSeq=0,tid=1,pid=30188,uid=0,mid=0,wordCount=0,byteCount=0,totalParameterCount=0,totalDataCount=0,maxParameterCount=10,maxDataCount=65535,maxSetupCount=0,flags=0x00,timeout=0,parameterCount=0,parameterOffset=0,parameterDisplacement=0,dataCount=0,dataOffset=0,dataDisplacement=0,setupCount=1,pad=0,pad1=0,searchAttributes=0x16,searchCount=200,flags=0x00,informationLevel=0x104,searchStorageType=0,filename=\]
        at jcifs.smb.SmbTreeImpl.checkRequest(SmbTreeImpl.java:519)
        at jcifs.smb.SmbTreeImpl.send(SmbTreeImpl.java:452)
        at jcifs.smb.SmbTreeConnection.send0(SmbTreeConnection.java:399)
        at jcifs.smb.SmbTreeConnection.send(SmbTreeConnection.java:314)
        at jcifs.smb.SmbTreeConnection.send(SmbTreeConnection.java:294)
        at jcifs.smb.SmbTreeHandleImpl.send(SmbTreeHandleImpl.java:130)
        at jcifs.smb.DirFileEntryEnumIterator1.open(DirFileEntryEnumIterator1.java:67)
        at jcifs.smb.DirFileEntryEnumIteratorBase.<init>(DirFileEntryEnumIteratorBase.java:67)
        at jcifs.smb.DirFileEntryEnumIterator1.<init>(DirFileEntryEnumIterator1.java:46)
        at jcifs.smb.SmbEnumerationUtil.doEnum(SmbEnumerationUtil.java:229)
        at jcifs.smb.SmbEnumerationUtil.listFiles(SmbEnumerationUtil.java:283)
        at jcifs.smb.SmbFile.listFiles(SmbFile.java:1192)
        at smbcli.main(smbcli.java:121)
[Thread-1] WARN jcifs.smb.SmbTransportImpl - Disconnecting transport while still in use Transport0[RT-AC68U<20>/192.168.0.27:445,state=5,signingEnforced=false,usage=1]: [SmbSession[credentials=GUEST,targetHost=rt-ac68u,targetDomain=null,uid=100,connectionState=2,usage=3], SmbSession[credentials=,targetHost=rt-ac68u,targetDomain=null,uid=101,connectionState=2,usage=0]]
[Thread-1] WARN jcifs.smb.SmbSessionImpl - Logging off session while still in use SmbSession[credentials=GUEST,targetHost=rt-ac68u,targetDomain=null,uid=100,connectionState=3,usage=3]:[SmbTree[share=MOVIES,service=?????,tid=0,inDfs=false,inDomainDfs=false,connectionState=0,usage=0]]
```
I would have expected not being able to browse root folder but apparently it is possible and get an SMBAuthException on the subfolders.
For full disclosure, MacOSX also have finder closing when accessing subfolders thus, it coud be a buggy server behavior.

**Comment by mbechler** [comment]
The exception happens because the share is reported as a IPC/Named Pipe type share, these cannot be listed. Packet capture would help to figure out what is happening there.
Being able to enumerate the shares without a valid user account is pretty normal (if that was a question).

**Comment by courville** [comment]
Sorry for the late reply (was away from the asus router). Please find the requested tcpdump pcap when getting the CIFSException while accessing movies share as GUEST on password protected share.
[cifsexception-asus-RT-AC68U-movies-guest.pcap.zip](https://github.com/AgNO3/jcifs-ng/files/2922096/cifsexception-asus-RT-AC68U-movies-guest.pcap.zip)

**Comment by mbechler** [comment]
From the trace it looks to me that accessing the movies share should produce a SmbAuthException as the server returns the correct error code. The other exception appears to be the result from actually trying to list files on the IPC$ share. But that does not really fit with the call/code. Can you provide a debug log as well?


> I would have expected not being able to browse root folder but apparently it is possible and get an SMBAuthException on the subfolders.
Just one additional remark here: The "root folder" is a completely different concept protocol wise from the things below it, so that behavior is somewhat expected.

**Comment by mbechler** [comment]
I think this may be fixed with 123ec077f976e92f69303b9951f8a137689212ba as well.

**Comment by courville** [comment]
I confirm that the issue is fixed on my side (was able to test during the holiday period). Thanks!
